### PR TITLE
mactop 0.1.7 (new formula)

### PR DESCRIPTION
**Pull request by metaspartan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by metaspartan** [comment]
Bummer, exactly 30 days old lol, will wait to resubmit tomorrow unless a maintainer can just merge in?

File: Formula/m/mactop.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  sha256 "647007fb085ad5cfe0b5283063931a671b21f583ca22bcab20465acedcc77776"
+  license "MIT"
+
+  depends_on "go" => :build
+  # Restrict formula to macOS on Apple Silicon only
+  depends_on arch: :arm64
+  depends_on :macos
+
+  def install
+    system "go", "build", *std_go_args
```
Comment (chenrui333) [inline Formula/m/mactop.rb:14]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w")
```

File: Formula/m/mactop.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Mactop < Formula
+  desc "Apple Silicon Monitor Top written in Go Lang"
+  homepage "https://github.com/context-labs/mactop"
+  url "https://github.com/context-labs/mactop/archive/refs/tags/v0.1.6.tar.gz"
+  sha256 "647007fb085ad5cfe0b5283063931a671b21f583ca22bcab20465acedcc77776"
+  license "MIT"
+
+  depends_on "go" => :build
+  # Restrict formula to macOS on Apple Silicon only
```
Comment (chenrui333) [inline Formula/m/mactop.rb:9]: no need to have the code comment as already in the `desc`

```suggestion
```

File: Formula/m/mactop.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Mactop < Formula
+  desc "Apple Silicon Monitor Top written in Go Lang"
```
Comment (chenrui333) [inline Formula/m/mactop.rb:2]:
```suggestion
  desc "Apple Silicon Monitor Top written in Golang"
```

**Comment by metaspartan** [comment]
Added in your suggestions @chenrui333 thank you!

Looks like its passing all CI now 👍 🔥

**Comment by chenrui333** [comment]
@metaspartan can you squash the commits and add the formula into autobump.txt, should be good after it.

**Comment by metaspartan** [comment]
> @metaspartan can you squash the commits and add the formula into autobump.txt, should be good after it.

Squashed and added!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@metaspartan, thanks for your first contribution to Homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**